l10n_it_website_sale_fiscalcode/controllers/main.py

```diff
@@ -22,15 +22,29 @@ def checkout_form_validate(self, mode, all_form_values, data):
         error, error_message = super().checkout_form_validate(
             mode, all_form_values, data
         )
-        partner_sudo = request.env.user.partner_id.sudo()
+        # Check fiscalcode
+        partner = request.env.user.partner_id
+        # company_type does not come from page form
+        company_type = partner.company_type
+        company_name = False
+        if "company_name" in data:
+            company_name = data.get("company_name")
+        else:
+            # when company_name is not posted (readonly)
+            if partner.company_name:
+                company_name = partner.company_name
+            elif partner.company_type == "company":
+                company_name = partner.name
         dummy_partner = request.env["res.partner"].new(
             {
                 "fiscalcode": data.get("fiscalcode"),
-                "is_company": partner_sudo.is_company,
+                "company_name": company_name,
```

Comment:
Review solo per il codice, da quanto leggo sopra, `company_name` potrebbe essere `False` qui, è ininfluente?

Reply:
Ciao @sergiocorato grazie per la review! Dovrebbe servire invece: da quello che ricordo questa PR ripristina il comportamento presente anche in v12, da cui ho sostanzialmente copiato il codice che era stato in parte erroneamente modificato bypassando la call al constrains check_fiscalcode().

[Anche su v12](https://github.com/OCA/l10n-italy/blob/e603afdd1e0da8bf7c2fd199ae2d4acf0985b9dd/l10n_it_website_portal_fiscalcode/controllers/main.py#L20) il `company_name` viene inizialmente istanziato a `False` ma se inviato tra i dati del SUBMIT viene poi valorizzato e aggiunto al dummy partner con cui si esegue la call. Dopodiché in check_fiscalcode [viene effettivamente valutato](https://github.com/OCA/l10n-italy/blob/652e7fad61b1cf66bbbc361fbd2dddb84022721c/l10n_it_fiscalcode/model/res_partner.py#L18) . Se non mi sono perso nient'altro direi che è giusto tenerlo così ?

Reply:
Potrebbe non arrivare valorizzato, e quindi arrivare `False` al constrains, ma in quel caso non ci sono problemi se è quello che intendevi

Reply:
Ottimo, non ho pratica col modulo quindi per me ok.